okapi-core/src/test/java/org/folio/okapi/InstallTest.java

```diff
@@ -1119,7 +1047,325 @@ public void installTenantInit(TestContext context) {
     Assert.assertTrue(
         "raml: " + c.getLastReport().toString(),
         c.getLastReport().isEmpty());
-    timerServer.close();
+    httpServerV1.close();
+  }
+
+  void createAsyncInitModule(TestContext context, String module) {
+    final String docModule = "{" + LS
+        + "  \"id\" : \"" + module + "\"," + LS
+        + "  \"name\" : \"async tenant init module\"," + LS
+        + "  \"provides\" : [ {" + LS
+        + "    \"id\" : \"_tenant\"," + LS
+        + "    \"version\" : \"2.0\"," + LS
+        + "    \"interfaceType\" : \"system\"," + LS
+        + "    \"handlers\" : [ {" + LS
+        + "      \"methods\" : [ \"POST\" ]," + LS
+        + "      \"pathPattern\" : \"/_/tenant\"," + LS
+        + "      \"permissionsRequired\" : [ ]" + LS
+        + "    }, {" + LS
+        + "      \"methods\" : [ \"GET\", \"DELETE\" ]," + LS
+        + "      \"pathPattern\" : \"/_/tenant/{id}\"," + LS
+        + "      \"permissionsRequired\" : [ ]" + LS
+        + "    } ]" + LS
+        + "  } ]," + LS
+        + "  \"requires\" : [ ]" + LS
+        + "}";
+
+    RestAssuredClient c = api.createRestAssured3();
+    c.given()
+        .header("Content-Type", "application/json")
+        .body(docModule).post("/_/proxy/modules").then().statusCode(201);
+  }
+
+  void deployAsyncInitModule(TestContext context, String module, int port) {
+    final String nodeDoc1 = "{" + LS
+        + "  \"instId\" : \"localhost-" + Integer.toString(port) + "\"," + LS
+        + "  \"srvcId\" : \"" + module + "\"," + LS
+        + "  \"url\" : \"http://localhost:" + Integer.toString(port) + "\"" + LS
+        + "}";
+
+    RestAssuredClient c = api.createRestAssured3();
+    c.given().header("Content-Type", "application/json")
+        .body(nodeDoc1).post("/_/discovery/modules")
+        .then().statusCode(201);
+    Assert.assertTrue("raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+  }
+
+  JsonObject enableAndWait(TestContext context, String tenant, String module) {
+    RestAssuredClient c = api.createRestAssured3();
+    Response r = c.given()
+        .header("Content-Type", "application/json")
+        .body("[ {\"id\" : \"" + module + "\", \"action\" : \"enable\"} ]")
+        .post("/_/proxy/tenants/" + tenant + "/install?async=true")
+        .then().statusCode(201)
+        .body(equalTo("[ {" + LS
+            + "  \"id\" : \""+ module + "\"," + LS
+            + "  \"action\" : \"enable\"" + LS
+            + "} ]"))
+        .extract().response();
+    Assert.assertTrue(
+        "raml: " + c.getLastReport().toString(),
+        c.getLastReport().isEmpty());
+    String locationInstallJob = r.getHeader("Location");
+
+    String path = locationInstallJob.substring(locationInstallJob.indexOf("/_/"));
```

Comment:
`indexOf("/_/")` is always 0.
```suggestion
    String path = r.getHeader("Location");
```

Reply:
LOL. Yes.. I was probably thinking that Location could be absolute URL!

Reply:
> LOL. Yes.. I was probably thinking that Location could be absolute URL!

Is this the location for the status of the install task in Okapi or the location of the status of an individual module upgrade?